lib/extractor/index.ts

```diff
@@ -203,27 +203,25 @@ function layersWithLatestFileModifications(
     // go over extracted files products found in this layer
     for (const filename of Object.keys(layer)) {
       // if finding a deleted file - trimming to its original file name for excluding it from extractedLayers
-      if (isDeletedFile(filename)) {
+      if (isWhitedOutFile(filename)) {
         removedFilesToIgnore.add(filename.replace(/.wh./, ""));
       }
 
-      // not adding the original file to extractedLayers
-      // and removing it from the set since it can be found in consecutive layers
+      // not adding previously found to be whited out files to extractedLayers
       if (removedFilesToIgnore.has(filename)) {
-        removedFilesToIgnore.delete(filename);
         continue;
       }
       if (
         Array.from(removedFilesToIgnore).some((removedFile) =>
-          isFileInFolder(removedFile, filename),
+          isFileInFolder(filename, removedFile),
         )
       ) {
         continue;
       }
       if (
-        // file was not found + avoid adding deleted files with .wh.
+        // file was not found in extractedLayers + avoid adding deleted files with .wh.
         !Reflect.has(extractedLayers, filename) &&
-        !isDeletedFile(filename)
+        !isWhitedOutFile(filename)
```

Comment:
I don't think we'd need this if we `continue` after checking if the file is whited-out